tutorials/integration.md

```diff
@@ -628,136 +708,93 @@ GSN started
 Relay is active, URL = http://127.0.0.1:44703 . Press Ctrl-C to abort
 ```
 1. Make sure you have the latest versions compiled and start the truffle console.
-+
 ```
 truffle compile
 truffle console
 ```
 1. Deploy the paymaster as explained above.
-+
-[source.javascript]
-```
+```javascript
 paymaster = await <paymaster contract>.new()
-paymaster.setRelayHub(<Relay hub address from gsn command>)
+paymaster.setRelayHub(<Relay hub address from the gsn command>)
+paymaster.setTrustedForwarder(<Forwarder address from the gsn command>)
 ```
 1. Fund the paymaster (it will transfer the ether to the relay hub).
-+
-[source.javascript]
-```
+```javascript
 paymaster.send(1e18)
 ```
 1. Deploy the target contract and configure the paymaster. If you are using
 `CaptureTheFlag` and `NaivePaymaster`, do this:
-+
-[source.javascript]
-```  
+```javascript
 flag = await CaptureTheFlag.new(<trusted forwarder address from gsn command>)
 paymaster.setTarget(flag.address)
 ```
-1. Configure the settings you need to use GSN, similar to what you did in the user 
-interface above. Remember to change `relayHubAddress`, `stakeManagerAddress`,
-and `preferredRelays` to the values you got from `gsn start`. 
-The `chainId` is set to `'*'` to accept
-any chain.
-+
-{% hint style="note" %}
-### NOTE:
-The gsnConfig definition should all be on one line. It is only separated here
-{% endhint %}
-for clarity.
-+
-[source.javascript]
-```
-Gsn = require("@opengsn/gsn/dist/src/relayclient/")
-RelayProvider = Gsn.RelayProvider
-configureGSN = require('@opengsn/gsn/dist/src/relayclient/GSNConfigurator').configureGSN
-ethers = require("ethers")
-
-gsnConfig = configureGSN({
-	paymasterAddress: paymaster.address,
-	gasPriceFactorPercent: 70,
-	methodSuffix: '_v4',
-	jsonStringifyRequest: true,
-	chainId: '*',
-	relayLookupWindowBlocks: 1e5,
-	relayHubAddress: '0xCA3ef05158d0dBD38cC7B49FbBc979d4cB977Ccc',
-	stakeManagerAddress: '0x316e9B4bBBBC9B585918CD19357fA686df636D22',
-	preferredRelays: [ 'http://127.0.0.1:43027' ],
-	verbose: true
-})    
-```
-1. Create the `Provider` object:
-+
-[source.javascript]
-```
-origProvider = web3.currentProvider
-gsnProvider = new RelayProvider(origProvider, gsnConfig)
-provider = new ethers.providers.Web3Provider(gsnProvider)
-```
+1. Configure the settings you need to use GSN, similar to what you did in 
+   the user interface above, and create the `Provider` object.
+
+   ```javascript
+   Gsn = require("@opengsn/gsn/dist")
+   ethers = require("ethers")
+   origProvider = web3.currentProvider
+   conf = {verbose: false}
+   conf.forwarderAddress = '<address of the forwarder from gsn start>'
```

Comment:
conf.forwarderAddress = require( '../build/gsn/Forwarder' ).address

Reply:
When I start GSN manually, it creates this file (`build/gsn/Forwarder.json`). But when I use the JavaScript `startGsn` , it does not get created.

Reply:
`loadDeployment()` is actually a better way to read the address than the `require( 'Forwarder').address`
calling `startGsn()` doesn't write the addresses to file, but returns them.
```
const { contractsDeployment } = await startGsn()
```
or:
```
const contractsDeployment = loadDeployment()
```
(don't write it, but currently our test env sucks and doesn't let you easily switch between the two options, so you need to select one and stick to it.)

Reply:
I would rather just show people both options, one for manual testing, the other for automated.

Reply:
both are automated: one is for starting GSN from within the test, and the other is for starting GSN just before the test.